barretenberg/cpp/src/barretenberg/ultra_honk/oink_prover.cpp

```diff
@@ -58,8 +58,10 @@ template <IsUltraFlavor Flavor> void OinkProver<Flavor>::prove()
     // Generate relation separators alphas for sumcheck/combiner computation
     proving_key->alphas = generate_alphas_round();
 
+#ifndef __wasm__
```

Comment:
what's the point of this? Seems like it won't be freed anyway if since ClientIVC is storing a shared_ptr to this.

I guess for UH in wasm, it will no longer be freed then

Reply:
You raise a good point--I wasn't thinking of UH WASM performance here. However, this optimization of freeing and reallocating the ck was only tested to work in x86, so you can think of this macro as almost being documentation of that fact. In the future we should have a PR that specifically tunes WASM memory performance of UH and yeah, perhaps an optimization at that point will require something different here.